src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/task/respondtoclaimcallbackhandlertasks/PopulateRespondentCopyObjects.java

```diff
@@ -0,0 +1,231 @@
+    private final IStateFlowEngine stateFlowEngine;
+    private final CoreCaseUserService coreCaseUserService;
+    private final UserService userService;
+    private final LocationReferenceDataService locationRefDataService;
+
+    public PopulateRespondentCopyObjects(ObjectMapper objectMapper,
+                                         CourtLocationUtils courtLocationUtils,
+                                         LocationReferenceDataService locationReferenceDataService,
+                                         IStateFlowEngine stateFlowEngine,
+                                         CoreCaseUserService coreCaseUserService,
+                                         UserService userService,
+                                         LocationReferenceDataService locationRefDataService) {
+
+        this.objectMapper = objectMapper;
+        this.courtLocationUtils = courtLocationUtils;
+        this.locationReferenceDataService = locationReferenceDataService;
+        this.stateFlowEngine = stateFlowEngine;
+        this.coreCaseUserService = coreCaseUserService;
+        this.userService = userService;
+        this.locationRefDataService = locationRefDataService;
+    }
+
+    @Override
+    public CallbackResponse execute(CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+        LocalDateTime dateTime = LocalDateTime.now();
+
+        CallbackResponse errorResponse = checkSubmitError(callbackParams, caseData, dateTime);
+        if (errorResponse != null) {
+            return errorResponse;
+        }
+
+        var isRespondent1 = findOutIsRespondent1(callbackParams);
+
+        List<LocationRefData> locations = fetchLocationData(callbackParams);
+
+        RequestedCourt requestedCourt1 = createRequestedCourt(locations, caseData);
+
+        CaseData.CaseDataBuilder updatedCaseData = updateCaseData(caseData, isRespondent1, requestedCourt1);
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(updatedCaseData.build().toMap(objectMapper))
+            .build();
+    }
+
+    private CallbackResponse checkSubmitError(CallbackParams callbackParams,
+                                              CaseData caseData,
+                                              LocalDateTime dateTime) {
+        // Show error message if defendant tries to submit response again
+        if (hasDefendantAlreadySubmitted(callbackParams, caseData)) {
+            return createErrorResponse(ERROR_DEFENDANT_RESPONSE_SUBMITTED);
+        }
+
+        // Show error message if defendant tries to submit a response after deadline has passed
+        if (isResponseSubmissionPassedDeadline(callbackParams, caseData, dateTime)) {
+            return createErrorResponse("You cannot submit a response now as you have passed your deadline");
+        }
+
+        return null;
+    }
+
+    private YesOrNo findOutIsRespondent1(CallbackParams callbackParams) {
```

Comment:
Allow my pettiness, but findOut is not a good name to check something. I'd set it either as a isRespondent1Present, checkRespondent1YesOrNo or simply checkRespondent1.